### PR TITLE
Wt 136 modal guidelines

### PR DESCRIPTION
**Pull request by rebelchris**
### Describe what this PR does
- Introducing the modal guidelines

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-136 #done

### CODE REVIEW

File: packages/shared/src/components/RankProgressWrapper.tsx [hunk 1]
```diff
@@ -86,7 +86,7 @@ export default function RankProgressWrapper({
           nextRank={nextRank}
         />
       )}
-      {shouldShowRankModal && (
+      {true && (
```
Comment (rebelchris) [inline packages/shared/src/components/RankProgressWrapper.tsx:89]: This one can't be right

**Review by rebelchris — COMMENTED** [review]
I'm not sure this is the right version, it seems only half-complete.

File: packages/shared/src/components/modals/Modal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  switch (variant) {
+    case ModalVariant.ResponsiveCenter:
+      return <ResponsiveModal {...props} />;
+    case ModalVariant.ResponsiveTop:
+      return <ResponsiveModal {...props} />;
+    case ModalVariant.FixedCenter:
+      return <StyledModal {...props} />;
+    default:
+      return <StyledModal {...props} />;
+  }
```
Comment (sshanzel) [inline packages/shared/src/components/modals/Modal/index.tsx:64]: Not sure, though In my mind, we will remove the usage of `ResponsiveModal` altogether, and create a stable equivalent of it.

**Comment by el** [comment]
Closing this PR, since I will create a new one

**Pull request closed without merging**